### PR TITLE
Update color variables on DSCO Button component

### PR DESCRIPTION
**Pull request by kelbyhawn**
Swaps out color variables from `shared/color.scss` with Primitive colors from `primitiveColors.css` on the Button component. 

👀 **DOTD note:** there might be Eyes diffs with this.

## Links
Jira ticket: [DESIGN2-200](https://codedotorg.atlassian.net/browse/DESIGN2-200)

## Testing story
Tested locally in Storybook

----

## Before - using `shared/colors.scss`
Before

## After - using `primitiveColors.css`
After

https://github.com/user-attachments/assets/b8019632-f534-4670-a6dc-13c8f4fbae6b

### CODE REVIEW

File: apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss [hunk 1]
```diff
@@ -53,66 +53,66 @@ a.button {
...
 }
 
 // Button Colors
 .button-primary,
 a.button-primary {
   &.button-purple {
-    background-color: $light_secondary_500;
-    color: $light_white;
+    background-color: var(--brand-purple-50);
+    color: var(--neutral-base-white);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss:65]: purple

File: apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss [hunk 2]
```diff
@@ -53,66 +53,66 @@ a.button {
...
+      background-color: var(--neutral-gray-20);
+      color: var(--neutral-base-white);
     }
   }
 
   &.button-black {
-    background-color: $light_black;
-    color: $light_white;
+    background-color: var(--neutral-base-black);
+    color: var(--neutral-base-white);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss:79]: black

File: apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss [hunk 3]
```diff
@@ -53,66 +53,66 @@ a.button {
...
+      background-color: var(--neutral-gray-20);
+      color: var(--neutral-base-white);
     }
   }
 
   &.button-white {
-    background-color: $light_white;
-    color: $light_black;
+    background-color: var(--neutral-base-white);
+    color: var(--neutral-base-black);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss:93]: white

File: apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss [hunk 4]
```diff
@@ -53,66 +53,66 @@ a.button {
...
+      background-color: var(--neutral-gray-80);
+      color: var(--neutral-gray-90);
     }
   }
 
   &.button-destructive {
-    background-color: $light_negative_500;
-    color: $light_white;
+    background-color: var(--sentiment-error-50);
+    color: var(--neutral-base-white);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/button/_baseButton/_baseButton.module.scss:107]: red

**Review by levadadenys — APPROVED** [review]
LGTM! Thank you!

One thing I also wonder though is that should we maybe add a comment somewhere that if there we'll be some failing eyes tests - dotd let us know? (there probably should be none, though it's possible there might be some and it might be ok and not)

**Pull request merged**